dali/operators/image/convolution/laplacian.cc

```diff
@@ -200,29 +187,42 @@ class LaplacianOpCpu : public OpImplBase<CPUBackend> {
 
 }  // namespace laplacian
 
+template <>
+bool Laplacian<CPUBackend>::ShouldExpand(const workspace_t<CPUBackend>& ws) {
+  const auto& input = ws.template Input<CPUBackend>(0);
+  auto layout = input.GetLayout();
+  dim_desc_ = convolution_utils::ParseAndValidateDim(input.shape().sample_dim(), layout);
+  bool should_expand = SequenceOperator<CPUBackend>::ShouldExpand(ws);
+  if (should_expand) {
+    assert(dim_desc_.usable_axes_start > 0);
+    dim_desc_.total_axes_count -= dim_desc_.usable_axes_start;
+    dim_desc_.usable_axes_start = 0;
+  }
```

Comment:
Initially I posted a comment, why `ShouldExpand` is not `const`, but here's an answer to that question :)

Feels kind of odd, that the method that tells, whether we should expand also has a side-effect. How about making the `ShouldExpand const` and moving the dim update to some other place?

Reply:
Hm, interesting point. The idea is that the `ShouldExpand` method is a place to 
1. decide if the "expansion mode" should kick off and 
2. for manual extra validation+processing of the initial layout.

The processing seems to be the contentious part. I wouldn't like to enforce separation of the two not to enforce parsing the layout twice (once in ShouldExpand for validation and once in SetupImpl). Moving extra validation to SetupImpl is not a good idea too because of error reporting - you don't have access to the initial layout there if the expansion kicked off, so you cannot report it to the user. Maybe just renaming ShouldExpand to SetupExpand would make it less confusing?